_posts/2024/2024-02-06-about-caching-in-eo.md

```diff
@@ -0,0 +1,241 @@
+These tasks happen one after the other, and each task relies on the output of the one before it.
+Each task has directories for input and output data, as well as a directory for storing cached data.
+Using the program name, each task can receive and store data.
+
+
+The previous caching mechanism in EO made use of distinct interfaces, specifically `Footprint` and `Optimization`.
+These caching interfaces shared similar logic, but with minor differences.
+For instance, `Footprint` verifies the EO version of the compiler, whereas the remaining checks are identical.
+Additionally, the conditions for searching data in the cache had errors. 
+The cached file is considered valid if the end time of goal's execution
+and the time of saving goal's result to the cache are equal.
+Due to this issue, the program behaved incorrectly, because saving the goal's result to the cache is not instantaneous.
+After conducting an in-depth analysis of the project's incorrect operation,
+several disadvantages of the previous caching mechanism in EO were brought to light:
+* Incorrect search conditions for data in the cache.
+* The verification method requires reading the file content, which results in inefficiencies.
+* The presence of multiple caching mechanisms creates challenges in identifying and rectifying caching errors.
+* Employing multiple caching mechanisms for similar entities is a suboptimal practice, 
+leading to redundancy and complicating the caching infrastructure.
+
+
+To address caching challenges in EO, we closely examined existing caching systems.
+Maven's caching mechanisms operate at the level of `phases` and individual project modules.
+However, we require a caching mechanism at the level of `goals`.
+Consequently, the existing caching systems in Maven do not align with our requirements for resolving present issues.
+Furthermore, we cannot use the `ccache` as the basis for creating caching in EO because `ccache` is a high-level tool
+and cannot work with individual compilation tasks.
+The concept of `Gradle Incremental build` bears resemblance to a tool that is essential for our purposes.
+It has the capability to manage separate compilation tasks based on inputs and outputs.
+However, an incremental build in Gradle may be redundant for the EO.
+In contrast to other programming languages, EO currently lacks pre-existing libraries that can be integrated 
+into the project.
+Consequently, there is no need to generate a fingerprint for each task's data. 
+Instead, it suffices to verify the last modification time of the files involved in EO compilation.
+The modification time of the preceding task must not exceed that of the subsequent one.
+As each task possesses directories for input and output data, accessing the desired file 
+via an absolute path enables retrieval of essential information, as file name and last modified time,
+from the file attributes without reading the file context.
+
+
+### Conclusion
```

Comment:
@Yanich96 I don't think we need such a conclusion in a blog post. It isn't a scientific article. Moreover it doesn't provide any useful information. Kinda "water".